internal/types/document_validation.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+	"strings"
+	"unicode/utf8"
+)
+
+// ValidationError describes an error that could occur when validating a document.
+type ValidationError struct {
+	Reason error
```

Comment:
```suggestion
	reason error
```

Basically, let's hide everything we can for now

Reply:
fixed